### DISCUSSION TITLE
Android VPN not functioning properly.

### DISCUSSION THREAD

**Discussion by saadullah2216** [strongswan/strongswan#1474] [Q&A]
I am working on an android browser application that includes VPN functionality. When I connect with a VPN and check my IP within my app with IP API it shows my actual IP and location however when I open google chrome it shows me a changed IP address which is the correct behavior.

when I connect to a VPN from some other app like Browsec and open my app IP API shows me the changed IP address which is perfectly fine.

**Comment by tobiasbrunner** [comment]
If you are using the strongSwan Android app source code as a base (note that it is [GPLv2 licensed](https://www.strongswan.org/license.html)) then the app itself is explicitly excluded from the VPN (allows the app to fetch CLRs/OCSP even if a TUN device exists that would block that traffic). See [here](https://github.com/strongswan/strongswan/blob/0c7bfec7af5ac0ebbec03ece027c143156a5821e/src/frontends/android/app/src/main/java/org/strongswan/android/logic/CharonVpnService.java#L1124-L1137).
